services/data_checker.js

```diff
@@ -0,0 +1,33 @@
+
+const mailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+
+const uuidRegex = /^([0-9a-fA-F]{8})-(([0-9a-fA-F]{4}\-){3})([0-9a-fA-F]{12})$/i
+
+const checkUUID = (uuid) => uuidRegex.test(uuid)
+
+const checkMail = (mail) => mailRegex.test(mail.toLowerCase())
+
+const Checker = {
+
+    checkDataWithUUID: (uuid,hostsMails,guestMail,duration) => {
```

Comment:
jak zwracasz stringa to on się nie zewaluuje zawsze do "true" ? Widziałem, że w serwisie używasz "!==" więc spoko, ale już widzę jak nas to kiedyś kopnie w dupe XD

Reply:
Dobre pytanie, ale z tego co widzę string się zawsze ewaluje do false XD. Ale z tego co teraz na szybko sprawdziłem to przy potrójnym równa się zawsze daje false dla porównania z każdą wartością logiczną. Przynajmniej na tej stronie:
https://playcode.io/new/

Na podstawie tego stwierdzam że przy potrójnym równa się nie powinno być problemu:
https://codeburst.io/javascript-double-equals-vs-triple-equals-61d4ce5a121a

Ale ogólnie niech się nasz mistrz js-owy wypowie @Davenury.

Reply:
Ogólnie działa to mniej więcej tak:
dowolna niepusta wartość (niepusty string, liczba różna od zera) jest wartością Truthy - czyli jak zrobicie `if(5)`, to wejdzie do ifa. Wartości castowane są wtedy do booleanów i zwracany jest im odpowiadający (wartość Truthy -> true, wartość Falsy -> false). Nie oznacza to jednak, że można sobie na spokojnie porównywać wartości różnych typów. O ile na przykład string może zcastować do number ('5' == 5 zwróci true), to wartość Truthy niekoniecznie będzie true. Nie wiem, czy Wam to cokolwiek pomogło, ale sam jakoś lepiej się nie znam XDDD

Reply:
No tak tylko że ja tam robię potrójne porównanie do true, co wymusza że typy muszą być takie same więc zawsze jak będzie string to to się wywali, więc wydaje mi się to na tyle bezpieczne o ile nie zrobimy tam optymalizacji i nie wywalimy potrójnego porównania XDD

Reply:
Tak, tak, tylko tu pytaliście, jak to działa, to odpowiedziałem XDDD

Reply:
No mi chodziło żebyś się wypowiedział czy takie rozwiązanie jakie jest teraz jest ok XDDD

Reply:
Wydaje mi się, że git, ale jakoś bardzo w jsa nie umiem, żeby mówić, co jest ok, a co nie xDDD